src/pages/TransactionsPage/TransactionsForm/TransactionsForm.js

```diff
@@ -0,0 +1,148 @@
+  const [amount, setAmount] = useInputState(0);
+  const [type, setType] = useInputState("income");
+  const [date, setDate] = useInputState("");
+  const [category, setCategory] = useState("select");
+  const [paymentType, setPaymentType] = useInputState("cash");
+  const [formErrors, setFormErrors] = useState({});
+  const [isTouched, setIsTouched] = useState({
+    title: false,
+    description: false,
+    amount: false,
+    date: false,
+    category: false,
+  });
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    console.log(formErrors);
+    // pass object to json-server or somewhere else, will add later
+    // closeModal();
+  };
+
+  const handleBlur = (field) => {
+    setIsTouched((values) => ({ ...values, [field]: true }));
+  };
+
+  const validateForm = () => {
+    const formValues = {
+      title: title,
+      description: description,
+      amount: amount,
+      date: date,
+      category: category,
+    };
+    const errors = validate(formValues);
+    setFormErrors(errors);
+  };
+
+  const firstUpdate = useRef(true);
+
+  useEffect(() => {
+    if (firstUpdate.current) {
+      firstUpdate.current = false;
+      return;
+    }
+    validateForm();
+  }, [title, description, amount, date, category]);
+
+  const handleCloseModal = () => {
+    // closeModal();
+  };
+
+  const paymentOptions = [
+    { id: "1-option", type: "cash" },
+    { id: "2-option", type: "card" },
+  ];
+  const paymentTypes = [
+    { id: "1-type", type: "income" },
+    { id: "2-type", type: "expense" },
+  ];
+
+  return (
+    <StyledTransactionForm onSubmit={handleSubmit}>
+      <Input
+        type="text"
+        label={"Title:"}
+        field={"title"}
+        value={title}
+        handleChange={setTitle}
+        errors={formErrors}
+        isTouched={isTouched}
+        handleBlur={handleBlur}
+      />
+      <Input
+        type="text"
+        label={"Description:"}
+        field={"description"}
+        value={description}
+        handleChange={setDescription}
+        errors={formErrors}
+        isTouched={isTouched}
+        handleBlur={handleBlur}
+      />
+      <Dropdown
+        options={["rent", "clothes", "food"]}
+        name={category}
+        field="category"
+        handleChange={setCategory}
+        label={"Category:"}
+        errors={formErrors}
+        isTouched={isTouched}
+        handleBlur={handleBlur}
+      />
+      <Input
+        type="number"
+        label={"Paid:"}
+        field={"amount"}
+        value={amount}
+        handleChange={setAmount}
+        errors={formErrors}
+        isTouched={isTouched}
+        handleBlur={handleBlur}
+      />
+      <DatePicker
+        field="date"
+        value={date}
+        handleChange={setDate}
+        errors={formErrors}
+        isTouched={isTouched}
+        handleBlur={handleBlur}
+      />
+      <div>
+        <RadioButton
+          options={paymentOptions}
+          field={"paymenType"}
+          label={"Paid by:"}
```

Comment:
i dont know if prettier adds some extra styling to your code or so, but remove those extra parentheses ;d